extensions/wikia/FacebookClient/FacebookClientController.class.php

```diff
@@ -126,17 +126,24 @@ private function base64UrlDecode( $input ) {
 	}
 
 	/**
-	 * Disconnect the user from Facebook
+	 * Disconnect the user from Facebook. This can occur in one of two ways, either when the user
+	 * deletes the Wikia App from facebook, of when they explicitly disconnect via Special:Preferences.
+	 * If it comes from Facebook, the request is internal and is sent by FacebookClientController::deauthorizeCallback.
+	 * If it comes explicitly from the user, the request is external and is sent by preferences.js::disconnect.
 	 *
-	 * @requestParam user This is a user object.  Only works for internal calls
+	 * @requestParam user This is a user object.
 	 */
 	public function disconnectFromFB() {
-		$user = $this->getVal( 'user', null );
 
-		if ( $user ) {
-			$user = User::newFromId( $user );
-		} else {
+		if ( $this->request->isInternal() ) {
+			$userId = $this->getVal( 'user' );
+			$user = User::newFromId( $userId );
```

Comment:
Ah, so that code was actually used to differentiate between internal and external requests. Before if the request was from preferences.js then it wouldn't have a 'user' value so the code would use the currently logged in user. If it came from deauthorizeCallback there wouldn't be a logged in user so it would pass in the user id and create the user object. That logic is now handled by lines 138 and lines 141.